One Response Model + Credentials as a Parameter + Request Models + Dockerfile

kangwork opened the pull request:
**Description**
Please refer to the commit messages below for more detailed descriptions. ;)
Note that all the changes made have been tested fully.

- **[Aggregated List]**
    - Studied and made the use of `aggregated_list` clearer and more efficient


- **[ABC]**
    - Studied ABC and made the Collector class as an abstract class
    - Made some simplifications


- **[Resource Model]**
    - Worked on consistency


- **[Response Model] ⭐** 
    - Defined a single response model that can be used in any APIs
    - Applied to all APIs and completed testing


- **[Credentials as Parameter] ⭐** 
    - Let users input the credentials information if wanted (otherwise find a pre-defined environmental variable if exists)
    - Worked on consistency + efficiency


- **[Style, Black Formatter, Makefile, Exception, etc.]**
    - See below

- **[Request Models] ⭐** 
    - Defined request models based on the updated specifications
    - Changed some credentials gathering logic + types accordingly

- **[Dockerfile] ⭐** 
    - Wrote a dockerfile
    - Changed some imports + type notations to incorporate the 3.8 compatibility

kangwork committed:
[Aggregated List] Checked that the list instances is returned as an empty list, so that we can just extend all the lists; also its performance was fine :)

kangwork committed:
[Collector as ABC] Made the Collector class as an abstract class

kangwork committed:
[Resource Models: __str__, Config.json_encoders] Removed unnecessary methods and attributes after checking no difference in output

kangwork committed:
[Response Model] Completed testing a new response model(1 for every apis)

kangwork committed:
[Black formatter]

kangwork committed:
[Makefile] Styling -- echo command not to be echoed

kangwork committed:
[Response Model] Fixed a bug where status code not being displayed on fastapi console + Removed unnecessary lines in the APIResponse model definition

kangwork committed:
[Style] Removed unnecessary imports

kangwork committed:
[Please Ignore] Backup before I remove the LogIn/LogOut features & secret_data dict(could not complete testing)

kangwork committed:
[Credentials as Parameter] Applied to main:list_all_resources api
- Tested

kangwork committed:
[Black Formatter]

kangwork committed:
[Credentials as Parameter] Applied to all APIs + Tested

kangwork committed:
[Credentials as Parameter] Only ask for required fields + Fill in missing required fields with default values if possible with warning messages

kangwork committed:
[Style] Removed an unused dict(all_credential_fields)

ImMin5 commented on src/main.py:
```diff
@@ -38,9 +38,9 @@ def read_root(request: Request):
+@app.post("/all-resources", response_model=APIResponse)
 @func_error_handler_decorator(logger=logger, is_api=True)
-def list_all_resources():
+def list_all_resources(credentials=Depends(get_credentials)):
```
사용자가 secret_data(dict) parameter로 입력 할 수 있게 변경 되면 좋을 것 같습니다.

kangwork replied:
확인하였습니다! 피드백 감사합니다!

ImMin5 commented on src/main.py:
```diff
@@ -38,9 +38,9 @@ def read_root(request: Request):
+@app.post("/all-resources", response_model=APIResponse)
 @func_error_handler_decorator(logger=logger, is_api=True)
-def list_all_resources():
+def list_all_resources(credentials=Depends(get_credentials)):
```
제가 예상 했던 모습은 아래와 같습니다!
![image](https://github.com/kangwork/MiniGoogleCloudCollector/assets/38625842/75e69cbb-9f49-440f-af3c-8c65270c435b)

kangwork replied:
넵! 그렇게 수정하도록 하겠습니다!

ImMin5 commented on src/models/response.py:
```diff
@@ -0,0 +1,25 @@
+    def __init__(self, **data):
+        super().__init__(**data, length=self.get_length(data, data))
+
+    def get_length(cls, v, values) -> int:
```
cls variable을 사용하신 이유가 있을 까요?

kangwork replied:
실수입니다.. 디자인 바꾸고 파라미터 변경을 놓쳤습니다.
덕분에 다시 공부할 수 있었습니다. 알려주셔서 감사합니다!

ImMin5 commented on src/models/response.py:
```diff
@@ -0,0 +1,25 @@
+from pydantic import BaseModel
+
+
+class APIResponse(BaseModel):
```
미팅때 APIResponse class에 대해서 간단하게 설명 부탁드립니다!

kangwork replied:
넵!

kangwork committed:
[Response Model] Removed an unused param + Changed cls --> self

kangwork committed:
[Credentials as Parameter] Make the user input be in form { secret_data: { FIELDS } }

kangwork committed:
[Makefile] Let clean remove ./__pycache__/*

kangwork committed:
[Dockerfile] Checked it is running; has some issues to fix though due to the python version

kangwork committed:
[Dockerfile] Replaced the type hinting with the older versions(typing) and few imports to be supported by python 3.8(older version)

kangwork committed:
[Black Formatter]

kangwork committed:
[Makefile] Added a line to set up an environment variable in Makefile in case a user wants to run via Makefile

kangwork committed:
[Response Model] Removed the unnecessary Config class (tested; there were no issues removing it)

kangwork committed:
[Request Model, Exception, Styles]
- [credentials.py] Changed the type/null checks as we have defined the request model + made the secret_data a required field (following the specifications)
- [exceptions.py] Defined CustomException to handle a case where we need to define an error code (e.g. Value Error, etc.) for better error displays

kangwork committed:
[Black Formatter]

kangwork committed:
[Request Model] Defined + applied + tested the model usage
- [collectors/iam_roles] Also changed the param type accordingly

kangwork committed:
[Style] Removed unnecessary imports

kangwork committed:
[Style] Removed an unused function in utils/logging.py + Renamed the private function

kangwork committed:
[Style] Made the ResourceAccessRequest as an abstract class as it was suggested in the former meeting (Lmk if not in this case 0_0!)"

kangwork committed:
[PIP Requirements] Removed the Annotated type notations --> No need for typing_extensions anymore :)

kangwork committed:
[Style] Removed unnecessary comments

kangwork committed:
[Response Models] Defined APIResponses & APIResponse according to the updated specifications
- Updated other modules accordingly to incorporate those changes

kangwork committed:
[Dockerfile] Fixed typing for python3.8 & relative path error

kangwork committed:
[Masking Sensitive Info] Masked the credentials info (Could not use __redacted__ feature)

kangwork committed:
[Dockerfile] Bring the log file to the host + Link the host's key file path to the docker's path using Docker Volume

kangwork committed:
[Style] Added placeholder files

kangwork committed:
[Black Formatter]

kangwork committed:
[Please Ignore] Ignore local changes to keys and logs

kangwork committed:
[Placeholder Files] Reupload the placeholder files

kangwork committed:
[Style] Delete keys/key.json

kangwork committed:
[Style] Removed the placeholder files (key.json, log.log)

kangwork committed:
[Encrypting Key File] Added a decrypt feature when using the mounted key file(which is assumed to be encrypted using the same passphrase)

kangwork committed:
[Logging] Fixed a bug; now if there is no log file, then it creates it

kangwork committed:
[DockerIgnore, GitIgnore, MntFolder] Added mounting folders

kangwork committed:
[Encrypting Key File, Dockerfile] Added a script that helps us encrypt the key file.
Updated dockerfile

kangwork committed:
Merge branch 'response-parameter' of https://github.com/kangwork/MiniGoogleCloudCollector into response-parameter
